distributed/http/worker/prometheus/core.py

```diff
@@ -161,6 +146,128 @@ def collect(self):
+
+        mean times per key:
+          pickle   = spill_time_total.pickle     / spill_count_total.disk_write
+          write    = spill_time_total.disk_write / spill_count_total.disk_write
+          unpickle = spill_time_total.unpickle   / spill_count_total.disk_read
+          read     = spill_time_total.disk_read  / spill_count_total.disk_read
+
+        mean bytes per key:
+          write    = spill_bytes_total.disk_write / spill_count_total.disk_write
+          read     = spill_bytes_total.disk_read  / spill_count_total.disk_read
+
+        mean bytes per second:
+          write    = spill_bytes_total.disk_write / spill_time_total.disk_write
+          read     = spill_bytes_total.disk_read  / spill_time_total.disk_read
+        """
+        try:
+            get_metrics = self.server.data.get_metrics  # type: ignore
+        except AttributeError:
+            return  # spilling is disabled
+        metrics = get_metrics()
+
+        total_bytes = CounterMetricFamily(
+            self.build_name("spill_bytes_total"),
+            "Total size of memory and disk accesses caused by managed data "
+            "since the latest worker restart",
+            labels=["event"],
+        )
+        # Note: memory_read is used to calculate cache hit ratios (see docstring)
+        for k in ("memory_read", "disk_read", "disk_write"):
+            total_bytes.add_metric([k], metrics[f"{k}_bytes_total"])
+
+        total_counts = CounterMetricFamily(
+            self.build_name("spill_count_total"),
+            "Total number of memory and disk accesses caused by managed data "
+            "since the latest worker restart",
+            labels=["event"],
+        )
+        # Note: memory_read is used to calculate cache hit ratios (see docstring)
+        for k in ("memory_read", "disk_read", "disk_write"):
+            total_counts.add_metric([k], metrics[f"{k}_count_total"])
+
+        total_times = CounterMetricFamily(
+            self.build_name("spill_time_total"),
+            "Total time spent spilling/unspilling since the latest worker restart",
+            labels=["event"],
+        )
+        for k in ("pickle", "disk_write", "disk_read", "unpickle"):
+            total_times.add_metric([k], metrics[f"{k}_time_total"])
+        yield total_times
+
+        max_times = GaugeMetricFamily(
+            self.build_name("spill_time_per_key_max"),
+            "Maximum time spent spilling/unspilling a single key "
+            "since the latest worker restart",
+            labels=["event"],
+        )
+        for k in ("pickle", "disk_write", "disk_read", "unpickle"):
+            max_times.add_metric([k], metrics[f"{k}_time_per_key_max"])
+        yield max_times
+
+        max_counts = GaugeMetricFamily(
+            self.build_name("memory_count_max"),
+            "Maximum number of keys in memory since the latest worker restart",
```

Comment:
Is this the max since last worker restart? Or max since last prometheus scrape? (and same question about other places that way "since the latest worker restart)

Reply:
Worker restart. prometheus scrapes are stricly read-only and idempotent.

Reply:
Oh, I see. #7360 introduced a max between polls. I guess I'll change mine to match.